src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -524,4 +534,96 @@ protected function getObject()
 
         return $obj;
     }
+
+    /**
+     * @test
+     */
+    public function testEncodeXmlWithBoolValue()
+    {
+        $expectedXml = '<?xml version="1.0"?>
+<response><foo>1</foo><bar>0</bar></response>
+';
+
+        $actualXml = $this->encoder->encode(array('foo' => true, 'bar' => false), 'xml');
```

Comment:
This is an actual behavior of XmlEncode, to cast bool to int. I am not sure that i should fix this non explicit behaviour, so i am just cover that with test.